scripts/prune-llvm-bin.py

```
@@ -0,0 +1,261 @@
+    }
+    manifest.write_text(json.dumps(data, indent=2))
+    print(f"Wrote manifest with {len(removed)} entries to {manifest}")
+
+
+def apply_manifest(manifest: Path, install_dir: Path) -> None:
+    if not manifest.is_file():
+        raise FileNotFoundError(f"Manifest not found: {manifest}")
+    data = json.loads(manifest.read_text())
+    removed = data.get("removed", [])
+    if not isinstance(removed, list):
+        raise ValueError("Manifest missing 'removed' list")
```

Ruff (0.14.8)

Prefer `TypeError` exception for invalid type

(TRY004)